pkg/services/publicdashboards/service/service.go

```diff
@@ -105,17 +105,24 @@ func (pd *PublicDashboardServiceImpl) FindByAccessToken(ctx context.Context, acc
 	return pubdash, nil
 }
 
+// FindEnabledPublicDashboardAndDashboardByAccessToken Gets public dashboard and a dashboard by access token if public dashboard is enabled
+func (pd *PublicDashboardServiceImpl) FindEnabledPublicDashboardAndDashboardByAccessToken(ctx context.Context, accessToken string) (*PublicDashboard, *dashboards.Dashboard, error) {
+	pubdash, dash, err := pd.FindPublicDashboardAndDashboardByAccessToken(ctx, accessToken)
```

Comment:
The common pattern is to check the error right after the call. You can get what you want with `if err != nil { return pubdash, dash, err}` right? Instead of adding a check of `err == nil` after the call.
If there is no error this will continue to the last return, in both cases returning the same of the original method. The only change will be if it is not enabled.